kratos/geometries/nurbs_curve_on_surface_geometry.h

```diff
@@ -0,0 +1,300 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Thomas Oberbichler
+//                   Tobias Teschemacher
+//                   Andreas Apostolatos
+//
+//  Ported from the ANurbs library (https://github.com/oberbichler/ANurbs)
+//
+
+#if !defined(KRATOS_NURBS_CURVE_ON_SURFACE_H_INCLUDED )
+#define  KRATOS_NURBS_CURVE_ON_SURFACE_H_INCLUDED
+
+// Project includes
+#include "geometries/geometry.h"
+
+#include "geometries/nurbs_curve_geometry.h"
+#include "geometries/nurbs_surface_geometry.h"
+#include "geometries/nurbs_shape_function_utilities/nurbs_curve_shape_functions.h"
+#include "geometries/nurbs_shape_function_utilities/nurbs_interval.h"
+
+
+namespace Kratos {
+
+template <int TWorkingSpaceDimension, class TCurveContainerPointType, class TSurfaceContainerPointType>
+class NurbsCurveOnSurfaceGeometry : public NurbsCurveGeometry<2, TCurveContainerPointType>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Geometry as base class.
+    // typedef Geometry<typename TCurveContainerPointType::value_type> BaseType;
+    typedef NurbsCurveGeometry<2, TCurveContainerPointType> BaseType;
+    typedef NurbsCurveOnSurfaceGeometry<TWorkingSpaceDimension, TCurveContainerPointType, TSurfaceContainerPointType> GeometryType;
+
+    typedef typename BaseType::IndexType IndexType;
+    typedef typename BaseType::SizeType SizeType;
+
+    typedef NurbsSurfaceGeometry<3, TSurfaceContainerPointType> NurbsSurfaceType;
+    typedef NurbsCurveGeometry<2, TCurveContainerPointType> NurbsCurveType;
+
+    /** Array of counted pointers to point. This type used to hold
+        geometry's points.*/
+    // typedef  typename TCurveContainerPointType PointsArrayType;
+    typedef  typename BaseType::PointsArrayType PointsArrayType;
+
+    typedef  typename BaseType::CoordinatesArrayType CoordinatesArrayType;
+
+    /// Counted pointer of NurbsCurveOnSurfaceGeometry
+    KRATOS_CLASS_POINTER_DEFINITION(NurbsCurveOnSurfaceGeometry);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    NurbsCurveOnSurfaceGeometry(
+        typename NurbsSurfaceType::Pointer pSurface,
+        typename NurbsCurveType::Pointer pCurve)
+        : BaseType(
```

Comment:
as we found another way in the projection and tesselation you can derive directly from geometry